src/pytopmod/core/dcel/operators.py

```diff
@@ -0,0 +1,277 @@
+
+    face_1_key = old_edge_node.face_1_key
+    face_2_key = old_edge_node.face_2_key
+
+    # 2 - Non-cofacial deletion.
+    if face_1_key != face_2_key:
+        # 2.1 - Update the face information.
+        # Create a new face.
+        new_face_key = mesh.create_face()
+
+        # Traverse face_1 and face_2, replace face_1 and face_2 with new_face.
+        for edge_key in list(
+            itertools.chain(face_trace(mesh, face_1_key), face_trace(mesh, face_2_key))
+        ):
+            edge_node = mesh.edge_nodes[edge_key]
+            if edge_node.face_1_key in (face_1_key, face_2_key):
+                edge_node.face_1_key = new_face_key
+            if edge_node.face_2_key in (face_1_key, face_2_key):
+                edge_node.face_2_key = new_face_key
+
+        # Delete face_1 and face_2.
+        mesh.delete_face(face_1_key)
+        mesh.delete_face(face_2_key)
+
+        # 2.2 - Update the edge information to delete the edge.
+        if vertex_1_previous_node.vertex_1_key == old_edge_node.vertex_1_key:
+            vertex_1_previous_node.vertex_1_next_key = vertex_1_next_key
+        else:
+            vertex_1_previous_node.vertex_2_next_key = vertex_1_next_key
+        if vertex_2_previous_node.vertex_1_key == old_edge_node.vertex_2_key:
+            vertex_2_previous_node.vertex_1_next_key = vertex_2_next_key
+        else:
+            vertex_2_previous_node.vertex_2_next_key = vertex_2_next_key
+
+        # Delete the edge.
+        mesh.delete_edge(old_edge)
+
+    # 3 - Cofacial deletion.
+    else:
+        # 3.1 - Update the edge information to delete the edge.
+        if vertex_1_previous_node.vertex_1_key == old_edge_node.vertex_1_key:
+            vertex_1_previous_node.vertex_1_next_key = vertex_1_next_key
+        else:
+            vertex_1_previous_node.vertex_2_next_key = vertex_1_next_key
+        if vertex_2_previous_node.vertex_1_key == old_edge_node.vertex_2_key:
+            vertex_2_previous_node.vertex_1_next_key = vertex_2_next_key
+        else:
+            vertex_2_previous_node.vertex_2_next_key = vertex_2_next_key
+
+        # Delete the edge.
+        mesh.delete_edge(old_edge)
+
+        # 3.2 - Update the face information.
+        # Create two new faces.
+        new_face_1_key = mesh.create_face()
+        new_face_2_key = mesh.create_face()
+
+        # Starting from vertex_1_previous, traverse the face and replace face_1 by
+        # new_face_1.
+        for edge_key in list(
+            face_trace(mesh, face_1_key, start_edge_key=vertex_1_previous)
+        ):
+            edge_node = mesh.edge_nodes[edge_key]
+            if edge_node.face_1_key == face_1_key:
+                edge_node.face_1_key = new_face_1_key
+            if edge_node.face_2_key == face_1_key:
+                edge_node.face_2_key = face_1_key
+
+        # Starting from vertex_2_previous, traverse the face and replace face_1 by
+        # new_face_2.
+        for edge_key in list(
```

Comment:
Suggestion: Identical code to the previous step, where the only diff is the starting edge, and what we replace it with

```python
# Starting from vertex_1_previous, traverse the face and replace face_1 by
# new_face_1.
replaceFace(mesh, face_1_key, new_face_1_key, start_edge_key=vertex_1_previous)

# Starting from vertex_2_previous, traverse the face and replace face_1 by
# new_face_2.
replaceFace(mesh, face_1_key, new_face_2_key, start_edge_key=vertex_2_previous)
```

Reply:
See previous comment